tls/s2n_ktls.c

```diff
@@ -54,13 +55,15 @@ static S2N_RESULT s2n_ktls_validate(struct s2n_connection *conn, s2n_ktls_mode k
     /* kTLS enable should only be called once the handshake has completed. */
     RESULT_ENSURE(is_handshake_complete(conn), S2N_ERR_HANDSHAKE_NOT_COMPLETE);
 
-    /* TODO support TLS 1.3
+    /* For now, only allow TLS1.3 for testing.
      *
-     * TLS 1.3 support requires sending the KeyUpdate message when the cryptographic
-     * key usage limits are met. However, this is currently only possible by applying a
-     * kernel patch to support this functionality.
+     * TLS1.3 is potentially more dangerous to enable than TLS1.2, since the kernel
+     * does not currently support updating TLS keys and therefore will fail if
+     * KeyUpdate messages are encountered.
```

Comment:
The control message in the self talk test isn't actually ignored, it's properly passed up to our post-handshake logic. So is the KeyUpdate message. But even if the KeyUpdate message was just swallowed by the kernel, the connection would still fail afterwards because the peer would update their sending key and we wouldn't update our receiving key.

My next PR deals with KeyUpdates: https://github.com/aws/s2n-tls/pull/4318